classes/migration/upgrade/v3_4_0/PreflightCheckMigration.php

```diff
@@ -223,10 +229,80 @@ protected function checkUsageStatsLogs(): void
         ) {
             throw new Exception("There are one or more log files that were unable to finish processing. This happens when the scheduled task to process usage stats logs encounters a failure of some kind. These logs must be repaired and reprocessed or removed before the upgrade can continue. The logs can be found in the folders reject, processing and stage in {$usageStatsDir}.");
         }
+
+        $this->checkUsageStatsLogFileUrl($filePathToCheck);
+    }
+
+    /**
+     * Check if the domains in the log file URLs are correct i.e. the same as the base_url in the config.inc.php.
+     * Only the first URL in the log file is checked (and the same assumed for all URLs in the log file).
+     *
+     * @throws Exception
+     */
+    protected function checkUsageStatsLogFileUrl(string $filePathToCheck): void
+    {
+        $fhandle = fopen($filePathToCheck, 'r');
```

Comment:
In general, I think it's better to use the class based `fopen` (`SplFileObject`), we probably also have some libraries to deal with files, but to initialize them doesn't seem to be worth, see: https://github.com/pkp/pkp-lib/blob/9764e3f2538df94211b30a1bf1a3110bcef24caa/classes/migration/upgrade/v3_4_0/I8073_RemoveNotesWithoutQueriesAndRelatedObjects.php#L38-L54).

Reply:
I will then use SplFileObject